rsciio/hamamatsu/_api.py

```diff
@@ -0,0 +1,529 @@
+                if time_range_units == "us":
+                    time_range_units = "µs"
+                detector["time_range"] = time_range
+                detector["time_range_units"] = time_range_units
+            else:
+                # TODO: add warning? only occurs for shading file
+                time_range = _str2numeric(time_range_str, "float")
+                detector["time_range"] = time_range
+        detector["acquisition_mode"] = AcqMode(int(acq_dict.get("AcqMode"))).name
+        return detector
+
+    def _map_spectrometer_md(self):
+        spectrometer = {}
+        spectro_dict = self.original_metadata.get("Comment", {}).get("Spectrograph", {})
+        try:
+            groove_density_str = spectro_dict["Grating"]
+        except KeyError:
+            groove_density = None
+        else:
+            groove_density_split = groove_density_str.split(" ")
+            if len(groove_density_split) == 2:
+                groove_density, groove_density_units = groove_density_str.split(" ")
+                groove_density = _str2numeric(groove_density, "int")
+                if groove_density_units != "g/mm":
+                    _logger.warning(
+                        f"groove_density is given in {groove_density_units}"
+                    )
+            else:
+                groove_density = groove_density_str
+        ## Remove grating when no unit (->1 or 2, but not lines per mm)
+        ## Same for blaze
+        ## warning for these cases?
+        if spectro_dict.get("Ruling") != "0" and spectro_dict.get("Blaze") != 0:
+            spectrometer["Grating"] = {
+                "blazing_wavelength": _str2numeric(spectro_dict.get("Blaze"), "float"),
+                "groove_density": _str2numeric(groove_density, "float"),
+            }
+        spectrometer["model"] = spectro_dict.get("DeviceName")
+        spectrometer["entrance_slit_width"] = _str2numeric(
+            spectro_dict.get("Side Ent. Slitw."), "float"
+        )  ## TODO: units?, side entry iris?
+        spectrometer["central_wavelength"] = _str2numeric(
+            spectro_dict.get("Wavelength"), "float"
+        )
+        return spectrometer
+
+    def map_metadata(self):
+        """Maps original_metadata to metadata."""
+        general = self._map_general_md()
+        signal = self._map_signal_md()
+        detector = self._map_detector_md()
+        spectrometer = self._map_spectrometer_md()
+
+        acquisition_instrument = {
+            "Detector": detector,
+            "Spectrometer": spectrometer,
+        }
+
+        metadata = {
+            "Acquisition_instrument": acquisition_instrument,
+            "General": general,
+            "Signal": signal,
+        }
+        _remove_none_from_dict(metadata)
+        return metadata
+
+
+def file_reader(filename, lazy=False, use_uniform_signal_axes=False, **kwds):
+    """Reads Hamamatsu's ``.img`` file.
+
+    Parameters
+    ----------
+    %s
+    %s
+
```

Comment:
Add `use_uniform_signal_axis` parameter, e.g. by copying the docstring from jobin-yvon (or by adding it to `_docstrings`)

Reply:
I copied it for now and will do a follow up PR, where I add this to _docstrings and move the signal axis fit to utils.